python/cudf/cudf/core/dataframe.py

```diff
@@ -3722,6 +3722,98 @@ def sort_values(
             keep_index=not ignore_index,
         )
 
+    def agg(self, aggs, axis=None):
+        """ 
+        Aggregate using one or more operations over the specified axis.
+
+        Parameters
+        ----------
+        aggs : str, list or dict
+            Function to use for aggregating data. Accepted combinations are:
+                string name (ex: "sum")
+                list of functions  (ex: ["sum", "min", "max"])
+                dict of axis labels specified operations per column
+                function name: must pass to DataFrame or DataFrame.apply
+
+        axis : {0 or ‘index’, 1 or ‘columns’}, default 0
+            If 0 or ‘index’: apply aggregation to each column
+            If 1 or ‘columns’: apply aggregation to each row.
+        
+        Returns 
+        -------
+        scalar : when Series.agg is called with single aggregation
+        Series : when DataFrame.agg is called with a single aggrgation
+        DataFrame : when DataFrame.agg is called with several aggregations
+
+        Notes
+        -----
+        Difference from pandas:
+          * Not supporting: *args, **kwargs
+
+        """
+        if axis is not None:
+            raise NotImplementedError("axis not implemented yet")
+
+        dtypes = [self[col].dtype for col in self._column_names]
+        common_dtype = cudf.utils.dtypes.find_common_type(dtypes)
+        df_normalized = self.astype(common_dtype)
+
+        if isinstance(aggs, list):
+            result = cudf.DataFrame()
+            # TODO : Allow simultaneous pass for multi-aggregation as a future optimization
+            for agg in aggs:
+                result[agg] = getattr(df_normalized, agg)()
+
+        elif isinstance(aggs, str):
+            result = cudf.DataFrame()
+            result[aggs] = getattr(df_normalized,aggs)()
+            result = result.T.loc[aggs]
+            result.name = None
+            return result
```

Comment:
I believe we need to more gracefully error in the situation someone provides an API that we don't support. I.E. in doing `pdf.agg('asdf')` pandas raises:
```
AttributeError: 'asdf' is not a valid function for 'DataFrame' object
```

Whereas we'll raise:
```
AttributeError: 'DataFrame' object has no attribute 'asdf'
```